qa/L0_kafka/kafka_test.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2018-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from math import prod
+import sys
+
+sys.path.append("../common")
+
+import os
+import sys
+import time
+import unittest
+import threading
+import numpy as np
+import infer_util as iu
+import test_util as tu
+from kafka import KafkaConsumer
+from kafka import KafkaProducer
+
+# By default, find tritonserver on "localhost", but can be overridden
+# with TRITONSERVER_IPADDR envvar
+_tritonserver_ipaddr = os.environ.get('TRITONSERVER_IPADDR', 'localhost')
+
+_trials = ["python"]
+_deferred_exceptions_lock = threading.Lock()
+_deferred_exceptions = []
+
+class KafkaTest(tu.TestResultCollector):
+    consumer_topics_ = []
+    producer_topic_ = ""
+    consumer_ = None
+    producer_ = None
+
+    def consume_requests(self):
+        # At this point, the test producer will have
+        # sent a message to the broker, so we must
+        # poll the broker, determine the position of
+        # the most recent message and use it.
+        partition = self.consumer_.assignment().pop()
+        self.consumer_.seek_to_end(partition)
+        end = self.consumer_.position(partition)
+        # Brand new topic partition
+        print(end)
+        if(end != 0):
+            self.consumer_.seek(partition, end-1)
```

Comment:
```suggestion
            # seek to latest message (end-1)
            self.consumer_.seek(partition, end-1)
```

Reply:
This is mentioned in the comment block just above